lib/warning.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+/**
+ * Class for storing warning details.
+ * @module fs-transform:warning
+ * @author Ryan Sandor Richards
+ */
+module.exports = Warning;
```

Comment:
Yes, these are not warnings that would be output to the command-line or something like that. We need to keep track of all the corner-case warnings so we can report them back to the UI and display meaningful messages to the user while she is adapting her repository to work with runnable.